### PR TITLE
interp: handle select clause

### PR DESCRIPTION
**Pull request by riacataquian**
the parser already supports the `select` command but the interpreter
treats it just like any other `for` command

given the syntax:
```
select name [in words ...]; do commands; done
```

fix support by:
* expanding and printing the words following `in`, to generate a list
  of items, each preceeded by a number and a close parenthesis
* display the prompt and read the line from stdin
* assigning the selected option to a special variable in Bash called
  `REPLY`
* support `PS3`, another special variable in Bash, which can be
  set to the prompt while the shell awaits for input. if not provided,
  defaults to `#?`:

additionally, `select` should loop until an input is provided. in case
of an invalid input, the name will be empty
```
$ bash -c 'PS3="pick one: "; select SEL in foo; do echo $REPLY; break; done'
1) foo
pick one:
1) foo
pick one:
1) foo
pick one: 1
```

fixes https://github.com/mvdan/sh/issues/969

### CODE REVIEW

File: interp/runner.go [hunk 1]
```diff
@@ -488,6 +501,80 @@ func (r *Runner) cmd(ctx context.Context, cm syntax.Command) {
...
+				for i, field := range items {
+					words[i] = word{
+						pos:   i + 1,
+						value: field,
+					}
+				}
+				// sort words by position
+				sort.Slice(words, func(i, j int) bool {
+					return words[i].pos < words[j].pos
+				})
```
[the remark this answers is not in the document]
Comment (riacataquian) [reply interp/runner.go:525]: oops. this was a remnant of an old code, as printing the menu gives me jumbled order. will remove.

File: interp/runner.go [hunk 2]
```diff
@@ -488,6 +501,80 @@ func (r *Runner) cmd(ctx context.Context, cm syntax.Command) {
...
+
+				readCh := make(chan []byte, 1)
```
[the remark this answers is not in the document]
Comment (riacataquian) [reply interp/runner.go:527]: agreed 🙈 thanks.

File: interp/runner.go [hunk 3]
```diff
@@ -488,6 +501,80 @@ func (r *Runner) cmd(ctx context.Context, cm syntax.Command) {
...
+							r.setVarString(name, words[c-1].value)
+							break
+						}
+					}
+				}
+
+				// execute commands until break or return is encountered
+				if r.loopStmtsBroken(ctx, x.Do) {
+					break
+				}
```
[the remark this answers is not in the document]
Comment (riacataquian) [reply interp/runner.go:575]: we don't:
```
$ ./gosh                                                                          
$ select SEL in foo; do echo $REPLY; break; done
1) foo
#?
1) foo
#?
1) foo
#?
1) foo
#?
1) foo
#? ^C
```
as per the loop above, we loop until choice is not empty.

File: interp/runner.go [hunk 4]
```diff
@@ -488,6 +501,80 @@ func (r *Runner) cmd(ctx context.Context, cm syntax.Command) {
...
+
+				ps3 := fmt.Sprintf("%s ", shellDefaultPS3)
+				if e := r.envGet(shellReplyPS3Var); e != "" {
+					ps3 = e
+				}
+
+				words := make([]word, len(items))
+				for i, field := range items {
+					words[i] = word{
+						pos:   i + 1,
```
[the remark this answers is not in the document]
Comment (riacataquian) [reply interp/runner.go]: thanks! had to do `choice-1` though as the index starts at 0.

File: interp/runner.go [hunk 5]
```diff
@@ -488,6 +501,80 @@ func (r *Runner) cmd(ctx context.Context, cm syntax.Command) {
 				items = r.fields(y.Items...) // for i in ...; do ...
 			}
 
+			if x.Select {
+				type word struct {
+					pos   int
+					value string
+				}
+
+				ps3 := fmt.Sprintf("%s ", shellDefaultPS3)
```
[the remark this answers is not in the document]
Comment (riacataquian) [reply interp/runner.go]: thanks for spotting. this actually is counter to bash implementation.

File: interp/runner.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (mvdan) [reply interp/runner.go:541]: I mean EOF for what it's worth, i.e. `^D` :) but looking at that edge case can always be a separate PR.

**Review by mvdan — APPROVED** [review]

**Pull request merged**